src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/Repository.kt

```diff
@@ -307,4 +310,16 @@ class Repository(
   fun deleteAllVisitDays() = agencyVisitDayRepository.deleteAll()
   fun deleteAllVisitTimes() = agencyVisitTimeRepository.deleteAll()
   fun <T> runInTransaction(block: () -> T) = block()
+
+  fun save(offenderNonAssociation: OffenderNonAssociation, offenderNonAssociationDetail: OffenderNonAssociationDetail): OffenderNonAssociation =
+    offenderNonAssociationRepository.save(offenderNonAssociation).apply {
+      offenderNonAssociationDetail.nonAssociation = this
+      offenderNonAssociationDetails.add(offenderNonAssociationDetail)
+    }
```

Comment:
Part of the reasoning behind a repository for each builder is to allow us to split up this God class... we're not there yet but this seems like a step backwards.

Reply:
Good point , deleted the save() method